### PR TITLE
feat: playlist page

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
- I've just added playlist feature to my video library.

**ONLY check the following files**

### Files to check
- [x] [src/App.js](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)

#### Components: 
- [x] [components/PlaylistModal.jsx](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-7b001d7ae2d46ce1b3465fbc10cd496a427d6601cd3acbccb3cc37b4ed0977fe)
- [x] [components/VideoCard.jsx](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-1461aba6ad23131521bb371ae9c7965c273560004ea4e2d74a3f5efe2b57ee1e)

#### Pages files

- [x] [pages/PlaylistFeed.js](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-0799c7ee5721572b8dc02d3fdd7cfcced4ee9614b6573308880f3ec6ad544592)
- [x] [pages/PlaylistVideos.js](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-2863da8eba5fe6e07119db99d524afcd834de895c1939e579a51e85af9f77486)
- [x] [reducers/playlistReducer.js](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-3e11f33b913ac8e0b6f58a14b12a874660cfd4c126f3bec457acf02fb755b30e)

 #### Contexts  + Reducers

- [x] [[contexts/ModalContext.js](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-fb409b3f2258749722385708524c8c4a896fa0595faa705ec15270afccdfc92b)
- [x] [contexts/PlaylistContext.js](https://github.com/nayyyhaa/inNotion-watch/compare/playlist-branch?expand=1#diff-fe5460bb9be5e866cf80f599065851d59a9ac3dd5bfe5324253a4dfc6b111549)
- [x] [/reducers/watchLaterReducer.js](https://github.com/nayyyhaa/inNotion-watch/compare/watchlater-branch?expand=1#diff-2901a823bd7fa89996c535dd055d30c5734c9b3509d8e87c962aa8fa3a6b03a1)

#### files to *ignore*
- [ ] data files
- [ ] CSS files
- [ ] index files

### CODE REVIEW

File: src/reducers/playlistReducer.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export const playlistReducer = (state, action) => {
+  // const index = state.findIndex((el) => el._id === action.payload._id);
```
Comment (anik31) [inline src/reducers/playlistReducer.js]: suggestion - avoid pushing commented code

**Review by anik31 — APPROVED** [review]
Nice work, the UI is good & everything is working perfectly. Code is well structured & clean.

suggestion - If a playlist has single video & it is deleted then the playlist also gets deleted. I think you can show that playlist is empty in this case.

File: src/reducers/playlistReducer.js (hunk 1 above)
Comment (nayyyhaa) [reply src/reducers/playlistReducer.js]: yes made the changes.

**Comment by nayyyhaa** [comment]
> Nice work, the UI is good & everything is working perfectly. Code is well structured & clean.
> 
> suggestion - If a playlist has single video & it is deleted then the playlist also gets deleted. I think you can show that playlist is empty in this case.

Yes, I thought of keeping functionality like "gallery in our phone" but, now i'm preserving the playlist if all videos are deleted.

**Review by vrx29 — APPROVED** [review]
Everything looks good. UI is clean. Just fix one thing there is a routing error if we click on in. notion watch logo in Navbar it shows page not found

**Comment by nayyyhaa** [comment]
> Everything looks good. UI is clean. Just fix one thing there is a routing error if we click on in. notion watch logo in Navbar it shows page not found

Yes made the changes.

**Comment by nayyyhaa** [comment]
Thanks for reviewing. Merging the PR.

**Pull request merged**